### PR TITLE
NP-3520: retrieve customer's cristin id from RequestInfos

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: apigateway/src/main/java/nva/commons/apigateway/RequestInfo.java [hunk 1]
```diff
@@ -78,32 +72,37 @@ public RequestInfo() {
...
     public String getQueryParameter(String parameter) throws BadRequestException {
-        return Optional.ofNullable(getQueryParameters().get(parameter))
-                   .orElseThrow(() -> new BadRequestException(MISSING_FROM_QUERY_PARAMETERS + parameter));
+        return getQueryParameterOpt(parameter)
+            .orElseThrow(() -> new BadRequestException(MISSING_FROM_QUERY_PARAMETERS + parameter));
     }
-    
+
+    @JsonIgnore
+    public Optional<String> getQueryParameterOpt(String parameter) {
```
Comment (axthosarouris) [inline apigateway/src/main/java/nva/commons/apigateway/RequestInfo.java:89]: Added a method for acquiring optional query parameters

**Review by brinxmat — APPROVED** [review]

File: apigateway/src/main/java/nva/commons/apigateway/RequestInfo.java [hunk 2]
```diff
@@ -194,55 +193,59 @@ public void setRequestContext(JsonNode requestContext) {
...
-    
+
     @JsonIgnore
     public Optional<String> getCustomerId() {
         return getRequestContextParameterOpt(CUSTOMER_ID);
     }
-    
+
+    @JsonIgnore
+    public Optional<URI> getCustomerCristinId() {
```
Comment (axthosarouris) [inline apigateway/src/main/java/nva/commons/apigateway/RequestInfo.java:222]: Added  a method for getting the CustomerCristinId

File: apigateway/src/test/resources/apiGatewayMessages/event_with_access_rights_claim.json [hunk 3]
```diff
@@ -49,7 +49,7 @@
         "email": "orestis.gkorgkas@unit.no",
         "custom:identifiers": "feide:og@unit.no",
         "email_verified": "false",
-        "custom:cristinId": "https://api.cristin.no/v2/institutions/20202",
+        "custom:cristinId": "https://example.com/someArbitrayCristinId234567890876545688",
```
Comment (axthosarouris) [inline apigateway/src/test/resources/apiGatewayMessages/event_with_access_rights_claim.json:52]: This did not need to change. but the line is not used either way or another

File: apigateway/src/test/resources/apiGatewayMessages/event_with_cristin_id.json [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        "sub": "01d3a3ac-6262-4e02-ba3e-e592d06e3ee6",
+        "custom:orgLegalName": "Unit",
+        "cognito:groups": "eu-west-1_ZgmwmNtMp_FeideIdentityProvider",
+        "custom:yearOfBirth": "1984",
+        "custom:orgName": "UNIT",
+        "iss": "https://cognito-idp.eu-west-1.amazonaws.com/eu-west-1_ZgmwmNtMp",
+        "custom:orgNumber": "NO919477822",
+        "custom:application": "NVA",
+        "custom:applicationRoles": "CURATOR,Curator,Creator",
+        "identities": "{\"dateCreated\":\"1605017657967\",\"userId\":\"fd42d7f46888c456162728642d6b2fd14bc0917f\",\"providerName\":\"FeideIdentityProvider\",\"providerType\":\"SAML\",\"issuer\":\"https:\\/\\/idp-test.feide.no\",\"primary\":\"true\"}",
```
Comment (axthosarouris) [inline apigateway/src/test/resources/apiGatewayMessages/event_with_cristin_id.json:43]: This is an AWS claim and I am thinking if we should just put the whole user object in the claims in a similar way. This way we avoid looking up users in the user service

**Pull request merged**